src/ray/raylet/node_manager.cc

```diff
@@ -517,16 +547,18 @@ void NodeManager::ProcessClientMessage(
       worker_pool_.RegisterWorker(std::move(worker));
       DispatchTasks();
     } else {
-      // Register the new driver.
-      JobID job_id = from_flatbuf(*message->driver_id());
-      worker->AssignTaskId(job_id);
+      // Register the new driver. Note that here the driver_id in RegisterClientRequest
```

Comment:
A couple lines above, we declare 

```cpp
auto worker = std::make_shared<Worker>(...);
```

but we already declared 

```cpp
auto worker = worker_pool_.GetRegisteredWorker(client);
```

earlier in this function (which should still be in scope). I'm a little surprised that this compiles, but I guess it's ok?

Reply:
This is allowed in C++, as long as variables with the same name reside in different "scopes". In this case the inner variable `worker` hides the outer variable in the switch scope.

https://stackoverflow.com/questions/4151203/why-is-this-c-working-variables-with-the-same-name

To avoid potential confusion, I renamed the outer `worker` variable to `registered_worker`.